### DISCUSSION TITLE
How to use path?

### DISCUSSION THREAD

**Discussion by Ted-juh** [RomanKubiak/ctrlr#148] [Q&A] [answered]
I have the following code:

`
cPath = Path()
cPath:addRectangle(100,100,50,50)
g:fillPath(cPath)
`

But it throws an error: attempt to call global "Path" (a nill value)

**Comment by dnaldoog** [comment, marked as answer]
<pre>
paint = function(--[[ CtrlrComponent --]] comp --[[ CtrlrComponent --]], g)
    g:fillAll(Colour(0xff0000ff))
    g:setColour(Colour(0xffffffff))
    local cPath = Path()
    local transform = AffineTransform()
    cPath:addRectangle(100, 100, 50, 50)
    g:fillPath(cPath, transform)
end

</pre>

* https://ctrlr.org/forums/topic/drawing-paths/#post-5293
* https://sourceforge.net/projects/jd990-super-editor/

**Comment by Ted-juh** [comment]
Need to get to work but after some fiddling, I found out that:

fillPath = "Path, AffineTransform"
strokePath = "Path, Stroke, AffineTransform"

So, when I left out the "stroke" it seemed to work.

**Reply by dnaldoog** [reply]
I just updated the comment using code from the JD-990 panel, but you beat me to it (_see above_)
